bin/nyc.js

```diff
@@ -117,7 +117,7 @@ if (process.env.NYC_CWD) {
   } else if (argv._.length) {
     // wrap subprocesses and execute argv[1]
     var nyc = (new NYC())
-    nyc.cleanup()
+    nyc.reset()
```

Comment:
I wonder if the `if (!process.env.NYC_CWD)` clause in `cleanup()` is causing some confusion here. I think it ensures child processes don't perform any cleanup, but maybe that guard should be in this file and not in the implementation.

TBH I'm not entirely sure what the expected behavior is and why it impacts `--all`.